code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -830,20 +830,28 @@ func (srh *StorageRestHandler) getReadPoolStat(w http.ResponseWriter, r *http.Re
 const cantGetConfigErrMsg = "can't get config"
 
 func getConfig(balances cstate.CommonStateContextI) (*Config, error) {
+	if *balances.GetConfig() != nil {
+		updateConfig(balances)
+	}
+	var conf = (*balances.GetConfig()).(Config)
+	return &conf, nil
+}
+
+func updateConfig(balances cstate.CommonStateContextI) error {
 	var conf = &Config{}
 	err := balances.GetTrieNode(scConfigKey(ADDRESS), conf)
```

Comment:
it looks like the only place where we are updating the balances.config is in (getConfig)[https://github.com/0chain/0chain/blob/4875edf53fa590cde20028fcb1a0d0ddd59ec696/code/go/0chain.net/smartcontract/storagesc/handler.go#L831-L832] 

But the ticket also tells to the update the config on block finalization also. should we also update there?

Reply:
@boddumanohar the SC will update the config if any, the 'update config on block finalization' is mean to update the config cache.